common/spring-cloud-stream-modules-common-configuration/src/main/java/org/springframework/cloud/stream/module/sftp/SftpSessionFactoryProperties.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2015 the original author or authors.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.module.sftp;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.stream.module.file.remote.AbstractRemoteServerProperties;
+
+/**
+ * @author Gary Russell
+ */
+@ConfigurationProperties
+public class SftpSessionFactoryProperties extends AbstractRemoteServerProperties {
```

Comment:
Add some jsr303 validation? some of the props have defaults of `""` instead of `null`, which seems to call for `@NotNull` at least